modules/gis/src/lib/binary-to-geojson.ts

```diff
@@ -156,7 +156,8 @@ function deduceReturnType(dataArray): 'FeatureCollection' | 'Geometry' {
 
 /** Parse input binary data and return an array of GeoJSON Features */
 function parseFeatureCollection(dataArray): Feature[] {
-  const features: Feature[] = [];
+  const featuresById: Map<number, Feature> = new Map<number, Feature>();
```

Comment:
If we know all the ids are covered, all we have to do is to create an array of the right length.
Suggested change:
  const featuresById: Map<number, Feature> = new Map<number, Feature>();
  const featuresById = new Array<Feature>(globalFeatureIds.length).fill(null);

This map business would seem to affect perf and memory.

Reply:
The issue is that we don't know the length until we've scanned through all the features, that's why I collect the features mapped to their original id and then create the array.

Reply:
Let's resolve this first, and then I can look at your other comments if they are still relevant